src/app/services/actions/snippet-action-creator.ts

```diff
@@ -86,3 +92,32 @@ export function getSnippet(language: string): Function {
     }
   };
 }
+
+export function constructHeaderString(sampleQuery: IQuery): string {
+  const { sampleHeaders, selectedVerb } = sampleQuery;
+  let headersString = '';
+  let isContentTypeInHeaders: boolean = false;
+  if (sampleHeaders && sampleHeaders.length > 0) {
+    const { constructedHeader, isContentTypeAvailable } = getHeaderStringProperties(sampleHeaders);
+    headersString = constructedHeader;
+    isContentTypeInHeaders = isContentTypeAvailable;
+  }
+
+  headersString += !isContentTypeInHeaders && selectedVerb !== 'GET' ? 'Content-Type: application/json\r\n' : '';
+  return headersString;
+}
+
+function getHeaderStringProperties(sampleHeaders: Header[]): IHeaderStringProperties {
+  let constructedHeader = ''
+  let isContentTypeAvailable: boolean = false;
+  sampleHeaders.forEach((header: Header) => {
+    isContentTypeAvailable = isContentTypeAvailable ? isContentTypeAvailable : checkIfContentTypeAvailable(header);
```

Comment:
Not sure that this will evaluate. 
It looks like it has already set isContentTypeAvailable to false so line 114 will always run the function for checkIfContentTupeAvailable(header)
Is running the check the intention inside the foreach?

Reply:
Yes. The intention is to run the check for each header until we find a content-type header. I think I can make it simpler with a .find() or .includes() array methods